src/components/molecules/TableHeader/TableHeader.tsx

```diff
@@ -36,20 +42,16 @@ const TableHeader: React.FC<TableHeaderProps> = ({
 
   const experience = useSelector(experienceSelector);
   const { recentVenueUsers } = useRecentVenueUsers();
+  const { isShown, show, hide } = useShowHide();
 
   const allTables = experience?.tables;
 
-  const tableOfUser = seatedAtTable
-    ? tables.find((table) => table.reference === seatedAtTable)
-    : undefined;
-
-  const usersAtCurrentTable = useMemo(
+  const tableOfUser = useMemo(
     () =>
-      seatedAtTable &&
-      recentVenueUsers.filter(
-        (user: User) => user.data?.[venueName]?.table === seatedAtTable
-      ),
-    [seatedAtTable, recentVenueUsers, venueName]
+      seatedAtTable
+        ? tables.find((table) => table.reference === seatedAtTable)
+        : undefined,
+    [seatedAtTable, tables]
   );
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Looks like this needs to be refactored into 2 `api` functions one for setting the lock status and one for leaving the table. I think this is OOS but should definitely get fixed. The update function itself can be removed because it would be hard to add type for the second param since it's dynamic and you can change anything in the database with it.

Reply:
If this is OOS has it been captured in a backlog issue+linked here? If it’s been fixed what’s the commit hash for it?

Reply:
Just created an issue: https://github.com/sparkletown/internal-sparkle-issues/issues/749